lib/stream.js

```diff
@@ -21,10 +21,21 @@
 
 'use strict';
 
+const {
+  ObjectDefineProperty,
+} = primordials;
+
+const {
+  promisify: { custom: customPromisify },
+} = require('internal/util');
+
 const pipeline = require('internal/streams/pipeline');
 const eos = require('internal/streams/end-of-stream');
 const internalBuffer = require('internal/buffer');
 
+// Lazy loaded
+let promises;
```

Comment:
Nit: 
```suggestion
let promises = null;
```
and then 
`if (promises === null)` in getters

to make this check simpler.